service/src/main/java/com/codeforcommunity/processor/TeamsProcessorImpl.java

```diff
@@ -35,4 +74,290 @@ public void disbandTeam(JWTData userData, int teamId) {
+          db.selectFrom(USERS_TEAMS.where(USERS_TEAMS.USER_ID.eq(memberId)
+          .and(USERS_TEAMS.TEAM_ROLE.eq(TeamRole.PENDING))))
+              .fetchOne();
+      if (approvedUserRecord != null) {
+        approvedUserRecord.setTeamRole(TeamRole.MEMBER);
+        approvedUserRecord.update();
+      } else {
+        throw new MemberStatusException(
+            memberId, teamId);
+      }
+    } else {
+      throw new WrongTeamRoleException(
+          teamId, TeamRole.MEMBER);
+    }
+  }
+
+  @Override
+  public void rejectUser(JWTData userData, int teamId, int memberId) {
+    checkUserExists(memberId);
+    checkTeamExists(teamId);
+    UsersTeamsRecord leaderTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(teamId)).and(USERS_TEAMS.TEAM_ROLE.eq(TeamRole.LEADER))
+            .fetchOne();
+    if (leaderTeamsRecord != null) {
+      UsersTeamsRecord approvedUserRecord =
+          db.selectFrom(USERS_TEAMS.where(USERS_TEAMS.TEAM_ID.eq(teamId).and(USERS_TEAMS.USER_ID.eq(memberId))
+          .and(USERS_TEAMS.TEAM_ROLE.eq(TeamRole.PENDING))))
+              .fetchOne();
+      if (approvedUserRecord != null) {
+        approvedUserRecord.setTeamRole(TeamRole.None);
+        approvedUserRecord.update();
+      } else {
+        throw new MemberStatusException(
+            userData.getUserId(), teamId);
+      }
+    } else {
+      throw new WrongTeamRoleException(
+          leaderTeamsRecord.getTeamId(), TeamRole.MEMBER);
+    }
+  }
+
+  @Override
+  public void kickUser(JWTData userData, int teamId, int memberId) {
+    checkUserExists(memberId);
+    checkTeamExists(teamId);
+    UsersTeamsRecord leaderTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(teamId)).and(USERS_TEAMS.TEAM_ROLE.eq(TeamRole.LEADER))
+            .fetchOne();
+    if (leaderTeamsRecord != null) {
+      UsersTeamsRecord approvedUserRecord =
+          db.selectFrom(USERS_TEAMS.where(USERS_TEAMS.USER_ID.eq(memberId)
+          .and(USERS_TEAMS.TEAM_ROLE.eq(TeamRole.MEMBER))))
+              .fetchOne();
+      if (approvedUserRecord != null) {
+        approvedUserRecord.setTeamRole(TeamRole.None);
+        approvedUserRecord.update();
+      } else {
+        throw new WrongTeamRoleException(teamId, TeamRole.MEMBER);
+      }
+    } else {
+      throw new WrongTeamRoleException(
+          teamId, TeamRole.LEADER);
+    }
+  }
+
+  @Override
+  public void leaveTeam(JWTData userData, int teamId) {
+    checkTeamExists(teamId);
+    UsersTeamsRecord usersTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(teamId))
+            .fetchOne();
+    if (usersTeamsRecord != null) {
+      if(usersTeamsRecord.getTeamRole() == TeamRole.LEADER) {
```

Comment:
we should allow the leader to leave the team if there is > 1 leader

Reply:
There is currently no way to reach a state with > 1 leader for a team.